client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-submit-button.tsx

```diff
@@ -0,0 +1,91 @@
+import { Button, FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import { useElements, CardElement } from '@stripe/react-stripe-js';
+import { useSelect } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:composite-checkout:credit-card' );
+
+export default function CreditCardSubmitButton( {
+	disabled,
+	onClick,
+	store,
+	stripe,
+	stripeConfiguration,
+	activeButtonText,
+} ) {
+	const { __ } = useI18n();
+	const fields = useSelect( ( select ) => select( 'credit-card' ).getFields() );
+	const useAsPrimaryPaymentMethod = useSelect( ( select ) =>
+		select( 'credit-card' ).useAsPrimaryPaymentMethod()
+	);
+	const cardholderName = fields.cardholderName;
+	const { formStatus } = useFormStatus();
+	const elements = useElements();
+	const cardElement = elements?.getElement( CardElement ) ?? undefined;
+
+	return (
+		<Button
+			// eslint-disable-next-line wpcalypso/jsx-classname-namespace
+			className="button is-primary"
+			disabled={ disabled }
+			onClick={ () => {
+				if ( isCreditCardFormValid( store, __ ) ) {
+					debug( 'submitting stripe payment' );
+					onClick( 'card', {
+						stripe,
+						name: cardholderName?.value,
+						stripeConfiguration,
+						cardElement,
+						useAsPrimaryPaymentMethod,
+					} );
+					return;
+				}
+			} }
+			buttonType="primary"
+			isBusy={ FormStatus.SUBMITTING === formStatus }
+			fullWidth
+		>
+			<ButtonContents formStatus={ formStatus } activeButtonText={ activeButtonText } />
```

Comment:
`ButtonContents` returns a string due to the fact that `__` returns a string. TS complains about this and says:
```
'ButtonContents' cannot be used as a JSX component.
  Its return type 'string' is not a valid JSX element.ts(2786)
```

Therefore, we can fix the issue by replacing the line with:
```
{ buttonContents( formStatus, activeButtonText ) }
```

Notice that I changed the word's capitalization since it is not a component anymore. Instead, it is a function that returns a string.

Reply:
Thanks for this suggestion! Fixed in ad7241beb664069a183a136c802882df5c9335ea